internal/server/boltdbstate/runner.go

```diff
@@ -302,9 +302,10 @@ func (s *State) runnerSetAdoptionState(
 }
 
 func (s *State) runnerOffline(dbTxn *bolt.Tx, memTxn *memdb.Txn, id string) error {
+	var r *pb.Runner
+
 	r, err := s.runnerById(dbTxn, id)
 	if status.Code(err) == codes.NotFound {
-		r = nil
```

Comment:
In the `NotFound` case when r is nil, won't the switch case on 323 still panic because `r.Kind`? Should we be returning here?

Reply:
You are right, I didn't double check to make sure that r.Kind was seeded with a default when initializing the struct and sure enough it doesn't.

Instead of initializing a blank runner struct i've short-circuited and returned nil in `83f900d76` as even if we get a NotFound error there isn't a runner we would need to delete, thus we can skip the rest of the function.